protocol/communities/persistence.go

```diff
@@ -1698,3 +1700,53 @@ func (p *Persistence) RemoveAllCommunityRequestsToJoinWithRevealedAddressesExcep
 		WHERE community_id = ? AND public_key != ?;`, communityID, pk, communityID, pk)
 	return err
 }
+
+func (p *Persistence) SaveCommunityShard(communityID types.HexBytes, shard *shard.Shard, clock uint64) error {
```

Comment:
suggestion: i think `dbClock >= clock` should not be an error. The return type can be changed to `(bool, error)`

Reply:
no, this must be an error. Imagine, that we have a handle function, which store the shard info and you got 2 msgs... first with clock 11 and second with clock 7. Clock 7 must not rewrite more fresh data

Reply:
I meant that message with `dbClock >= clock` should still be ignored, but no error raised because older message can arrive after fresh one and it's kind of normal

Reply:
ok, understood, in the HandleCommunityShardInfo I'm not returning an error if error is an outdated clock
P.S. Added as a separate commit